src/main/java/org/opentripplanner/routing/stoptimes/StopTimesHelper.java

```diff
@@ -55,8 +57,20 @@
       startTime = System.currentTimeMillis() / 1000;
     }
     List<StopTimesInPattern> result = new ArrayList<>();
-    Date date = new Date(startTime * 1000);
-    ServiceDate[] serviceDates = {new ServiceDate(date).previous(), new ServiceDate(date), new ServiceDate(date).next()};
+
+    LocalDate date = LocalDate.ofEpochDay(startTime/ONE_DAY_SECONDS);
```

Comment:
This is still wrong when calculating DST changes, it should use the `Instant` or some other method of converting from a unix timestamp to a local date with timezone